src/routes/github/github-oauth-router.ts

```diff
@@ -13,12 +13,18 @@ import { GitHubServerApp } from "models/github-server-app";
 const logger = getLogger("github-oauth");
 const appUrl = envVars.APP_URL;
 const scopes = ["user", "repo"];
-const callbackPath = "/callback";
+const callbackSubPath = "/callback";
+const callbackPathCloud = `/github${callbackSubPath}`;
+const callbackPathServer = `/github/<uuid>${callbackSubPath}`;
+
+const getRedirectUrl = async (res, state) => {
+	let callbackPath = callbackPathCloud;
+	if (res.locals?.gitHubAppConfig?.uuid) {
+		callbackPath = callbackPathServer.replace("<uuid>", res.locals.gitHubAppConfig.uuid);
+	}
```

Comment:
Made this change to be explicit what kind of URLs we support, which is only 2:
<img width="1792" alt="image" src="https://user-images.githubusercontent.com/20631664/204436022-fac02379-0dd4-4b21-908f-43a23d6b700a.png">

Given that `GitHubAuthMiddleware` middleware can be used in many places, we cannot rely on `req` anymore...